### ISSUE TITLE
[LINT_BUG]: Using R6 public attribute should not lint

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#80]
### box.linters version

0.0.0.9013

### Sample source code to lint
Code from function example

```R
box::use(
  R6[R6Class],
)

goodClass <- R6Class('goodClass',
  public = list(
    public_attr,
    initialize = function() {
      private$private_func()
    },
    some_function = function () {
      private$private_attr
    }
  ),
  private = list(
    private_attr = 'private data',
    private_func = function() {
      self$public_attr
    }
  )
)
```
### Lint command used
```R
lintr::lint(
  code,
  linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters)
)
```

### Lint result

```
<text>:19:7: warning: [r6_usage_linter] Internal object call not found.
      self$public_attr
      ^~~~~~~~~~~~~~~~
```

### Expected result

Accessing `self$public_attr` should not lint
